getting members from non existing collection returns empty string instead of json

https://github.com/phaidra/phaidra-api/blob/f8d8d59820784c40907f601ad6bfd04b7016c2a1/PhaidraAPI/Controller/Collection.pm#L295

for example:
```bash
$ curl -X GET "https://phaidratest.cab.unipd.it/api/collection/1/members"
$ 
$ curl -X GET "https://services.phaidra.univie.ac.at/api/collection/o:23213123/members"
$
```

the call return a json for non collection content types:

```bash
$ curl -X GET "https://services.phaidra.univie.ac.at/api/collection/o:1433423/members"
{"alerts":[],"metadata":{"members":[]}}$
```

the call should return at least an empty json ({}) but would be better to return:
```json
{"alerts":[],"metadata":{"members":[]}}
```

Related log messages:
```
[2022/04/15 10:34:34] [ERROR] [6224] [error] Collection->get_members: Cannot get lastModifiedDate!
[2022/04/15 10:34:34] [ERROR] [6224] [error] must specify key at /usr/local/phaidra/phaidra-api/PhaidraAPI/Model/Collection.pm line 217.
```

--- a/PhaidraAPI/Controller/Collection.pm
+++ b/PhaidraAPI/Controller/Collection.pm
@@ -289,6 +289,11 @@ sub get_collection_members {
   my $pid     = $self->stash('pid');
   my $nocache = $self->param('nocache');
 
+  unless (defined($pid)) {
+    $self->render(json => {alerts => [{type => 'danger', msg => 'Undefined pid'}]}, status => 400);
+    return;
+  }
+
   my $coll_model = PhaidraAPI::Model::Collection->new;
   my $res        = $coll_model->get_members($self, $pid, $nocache);
 

--- a/PhaidraAPI/Model/Collection.pm
+++ b/PhaidraAPI/Model/Collection.pm
@@ -126,8 +126,31 @@ sub get_members {
 
   my $search_model = PhaidraAPI::Model::Search->new;
 
+  my $cmodel;
+  my $res_cmodel   = $search_model->get_cmodel($c, $pid);
+  if ($res_cmodel->{status} ne 200) {
+    $self->app->log->error("Collection->get_members: pid[$pid] could not get cmodel");
+    return $res_cmodel;
+  }
+  else {
+    $cmodel = $res_cmodel->{cmodel};
+  }
+
+  unless ($cmodel) {
+    $c->app->log->error("Collection->get_members: pid[$pid] Undefined cmodel");
+    unshift @{$res->{alerts}}, {type => 'danger', msg => 'Object content model is undefined'};
+    $res->{status} = 400;
+    return $res;
+  }
+
+  if ($cmodel ne 'Collection') {
+    $c->app->log->error("Collection->get_members: pid[$pid] cmodel[$cmodel] is not Collection");
+    unshift @{$res->{alerts}}, {type => 'danger', msg => 'Object content model is not Collection'};
+    $res->{status} = 400;
+    return $res;
+  }
+
   # get lastModifiedDate and check if the members are cached
-  my $lmd;
   my $cachekey;
   my $cached_members;
   my $r = $search_model->get_last_modified_date($c, $pid);
@@ -136,11 +159,11 @@ sub get_members {
   }
   else {
     $c->app->log->error("Collection->get_members: Cannot get lastModifiedDate!");
+    return $r;
   }
-  if ($cachekey) {
-    $cached_members = $c->app->chi->get($cachekey);
-  }
-
+  
+  $cached_members = $c->app->chi->get($cachekey);
+  
   $nocache = $nocache ? $nocache : 0;
   if ($cached_members && ($nocache != 1)) {
     $c->app->log->debug("[cache hit] $cachekey");
@@ -218,7 +241,7 @@ sub get_members {
     $c->app->chi->set($cachekey, $cached_members, '1 day');
   }
 
-  # we have to return mempty array if there's is nothing
+  # we have to return an empty array if there is nothing
   unless (defined($cached_members)) {
     my @arr = ();
     $res->{members} = \@arr;

